source/Calamari.Azure/Integration/Websites/Publishing/ResourceManagerPublishProfileProvider.cs

```diff
@@ -43,12 +43,25 @@ public static SitePublishProfile GetPublishProperties(string subscriptionId, str
                     }
                 }
 
-                var matchingSite = sites
-                    .FirstOrDefault(webApp => string.Equals(webApp.Name, azureTargetSite.Site, StringComparison.CurrentCultureIgnoreCase) &&
-                                         (string.IsNullOrWhiteSpace(resourceGroupName) || string.Equals(webApp.ResourceGroup, resourceGroupName, StringComparison.InvariantCultureIgnoreCase)));
+                var matchingSites = sites
+                    .Where(webApp => string.Equals(webApp.Name, azureTargetSite.Site, StringComparison.CurrentCultureIgnoreCase) &&
+                                         (string.IsNullOrWhiteSpace(resourceGroupName) || string.Equals(webApp.ResourceGroup, resourceGroupName, StringComparison.InvariantCultureIgnoreCase)))
+                                         .ToList();
 
-                if (matchingSite == null)
+                if (!matchingSites.Any())
                     throw new CommandException($"Could not find Azure WebSite '{azureTargetSite.Site}' in subscription '{subscriptionId}'");
+                
+                // if more than one site, fail
+                if (matchingSites.Count > 1)
+                    throw new CommandException(
+                        $"Found {matchingSites.Count} matching the site name '{azureTargetSite.Site}' in subscription '{subscriptionId}'.{(string.IsNullOrWhiteSpace(resourceGroupName) ? " Please supply a Resource Group name." : string.Empty)}");
+
+                var matchingSite = matchingSites.First();
+
+                if (string.IsNullOrWhiteSpace(resourceGroupName))
+                {
+                    resourceGroupName = matchingSite.ResourceGroup;
```

Comment:
Should we just always do this assignment? It'll always give you the right answer, and 1 less `if` statement is always a good thing, right? :)

Reply:
similar to your comment about incorrect modifications in earlier code, if the matching site query gets borked, then this will ensure that the resource group supplied will take precedence